File: codes/python/chap06/answer_A41.py
# 入力
N = int(input())
S = input()

# 答えを求める
Answer = False
for i in range(0, N-2):
	if S[i] == 'R' and S[i+1] == 'R' and S[i+2] == 'R':
		Answer = True
	if S[i] == 'B' and S[i+1] == 'B' and S[i+2] == 'B':
		Answer = True

# 出力
if Answer == True:
	print("Yes")
else:
	print("No")

File: codes/cpp/chap06/answer_A41.cpp
#include <iostream>
using namespace std;

int main() {
	// 入力
	int N; char S[200009];
	cin >> N;
	for (int i = 1; i <= N; i++) cin >> S[i];

	// 答えを求める
	bool Answer = false;
	for (int i = 1; i <= N - 2; i++) {
		if (S[i] == 'R' && S[i + 1] == 'R' && S[i + 2] == 'R') Answer = true;
		if (S[i] == 'B' && S[i + 1] == 'B' && S[i + 2] == 'B') Answer = true;
	}

	// 出力
	if (Answer == true) cout << "Yes" << endl;
	else cout << "No" << endl;
	return 0;
}
